airbyte-integrations/connectors/destination-mssql-v2/src/test-integration/kotlin/io/airbyte/integrations/destination/mssql/v2/MSSQLWriterTest.kt

```diff
@@ -109,40 +128,138 @@ class MSSQLDataDumper : DestinationDataDumper {
+    private val shouldCleanUp: Boolean,
+    private val mssqlSpecification: MSSQLSpecification?,
+    private val configProvider: (MSSQLSpecification) -> MSSQLConfiguration
+) : DestinationCleaner {
+
+    private val dateFormatter = DateTimeFormatter.ofPattern("yyyyMMdd")
+
+    override fun cleanup() {
+        if (!shouldCleanUp || mssqlSpecification == null) return
+
+        // Cleanup azure blobs if in BulkLoad configuration
+        cleanupAzureBlobsIfNeeded()
+
+        // Cleanup older test schemas
+        cleanupOldTestSchemas()
+    }
+
+    /** Cleans up blobs older than 1 hour if the load configuration is [BulkLoadConfiguration]. */
+    private fun cleanupAzureBlobsIfNeeded() {
+        val loadConfig = mssqlSpecification!!.toLoadConfiguration().loadTypeConfiguration
+        if (loadConfig !is BulkLoadConfiguration) return
+
+        val azureBlobClient = AzureBlobStorageClientCreator.createAzureBlobClient(loadConfig)
+        runBlocking {
+            val blobList = mutableListOf<AzureBlob>()
+            azureBlobClient.list("").toList(blobList)
+
+            for (blobItem in blobList) {
+                val properties = azureBlobClient.getProperties(blobItem.key)
+                properties?.let { createdTime ->
+                    val hoursSinceCreation =
+                        ChronoUnit.HOURS.between(createdTime, OffsetDateTime.now())
+                    if (hoursSinceCreation >= 1) {
+                        azureBlobClient.delete(blobItem.key)
+                    }
+                }
             }
-        return MSSQLConfigurationFactory()
-            .makeWithOverrides(spec = spec as MSSQLSpecification, overrides = configOverrides)
+        }
     }
-}
 
-class MSSQLDataCleaner : DestinationCleaner {
-    override fun cleanup() {
-        // TODO("Not yet implemented")
+    /**
+     * Removes any schema whose name is of the form `testYYYYMMDDsomeSuffix` and is exactly one day
+     * old (compared to the current local date).
+     */
+    private fun cleanupOldTestSchemas() {
+        val config = configProvider(mssqlSpecification!!)
+        val dataSource = DataSourceFactory().dataSource(config)
+
+        dataSource.connection.use { conn ->
+            try {
+                val query = "SELECT name FROM sys.schemas WHERE name LIKE 'test%'"
+                val todayMinusOne = LocalDate.now().minusDays(1)
+
+                conn.createStatement().use { stmt ->
+                    val rs = stmt.executeQuery(query)
+                    val schemasToDelete = mutableListOf<String>()
+
+                    while (rs.next()) {
+                        val schemaName = rs.getString("name") ?: continue
+
+                        // Ensure the schema name can contain "test" + 8-digit date => length >= 12
+                        if (schemaName.length >= 12) {
+                            val datePart = schemaName.substring(4, 12)
+                            if (datePart.matches(Regex("\\d{8}"))) {
```

Comment:
fyi I think this logic already exists (IntegrationTest.randomizedNamespaceRegex / isNamespaceOld) - https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/IntegrationTest.kt#L313

example usage in iceberg https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/bulk/toolkits/load-iceberg-parquet/src/testFixtures/kotlin/io/airbyte/cdk/load/data/icerberg/parquet/IcebergDestinationCleaner.kt#L23-L26

(and if what you're doing is different, then we should add a comment explaining the difference)

Reply:
done